### ISSUE TITLE
Consider making a failure to upload cause the travis build to fail

### ISSUE THREAD

**Issue by solvingj** [conan-io/conan-package-tools#111]
We had a few builds passing on travis, and only found out later that they weren't getting uploaded due to bad credentials.  Probably the best way to introduce something like this would be to introduce another parameter+environment_variable pair to toggle "UPLOAD_FAILURE_FATAL" or something.

**Comment by uilianries** [comment]
Or maybe, when you want to upload the package, the credentials could be checked before to start the build.

**Comment by solvingj** [comment]
that seems like a FANTASTIC idea!

**Comment by uilianries** [comment]
What do you think @lasote and @memsharded? This feature is pretty simple and could save time when anyone want to push a package. In my experience, I saw this situation some times, when I forgot to set my CONAN_PASSWORD in a new project, or I use an incorrect variable.

**Comment by lasote** [comment]
Yes, sounds good!

**Comment by lasote** [comment]
But, about the `UPLOAD_FAILURE_FATAL` is not dangerous to finish a build with green status and with the upload failing?

**Comment by solvingj** [comment]
That's what is happening to us right now when the cause is "Missing information" (for example, conan password was not present).

**Comment by lasote** [comment]
But in that case the builds fail, right?

**Comment by lasote** [comment]
Oh! so it's not failing. It should be definitely added opt-out or mandatory

**Comment by solvingj** [comment]
And since I failed to include the actual output from the logs in my original report, here they are for posterity, i just happened across one. 

```
Upload not possible, some parameter (reference, password or channel) is missing!
Build success
```

**Comment by lasote** [comment]
@solvingj Thanks to @uilianries now we have an opt-out parameter `skip_check_credentials` that will check the credentials at the beginning or fail. It has been released in 0.5.7.
Let me know how it works!

**Comment by lasote** [comment]
I'm adding now that, if you specify an upload URL and some of the parameters (reference, password or channel) are missing, it raises an error.

**Comment by lasote** [comment]
Released 0.5.8. Now with a configured upload remote and missing parameter should fail. Please check and reopen if needed
